interfaces/core.go

```diff
@@ -184,6 +184,20 @@ type Interface interface {
 	// unambiguous connection candidate and declaration-based checks
 	// allow.
 	AutoConnect(plug *snap.PlugInfo, slot *snap.SlotInfo) bool
+
+	// UpdateStaticAttrs returns whether or not, on refresh, static attributes should
+	// and can be updated.
+	UpdateStaticAttrs(plug *snap.PlugInfo, slot *snap.SlotInfo) UpdateStaticAttrsResult
+}
+
+// UpdateStaticAttrsResult is the value indicating whether or not, on refresh, static attributes
+// should and can be updated. Should indicates that the updates should be updated. Can indicates
+// whether not not there is some incompatible change that would prevent them from being updated
+// (e.g. a change introducing a mismatch between the content value in a content interface.)
+// If Should is false, Can is ignored.
```

Comment:
why just not return one boolean?

Reply:
Yeah the dual bools are to preserve the existing logging/error behavior (most interfaces were not touched at all, but a couple of them could be updated in a way where the updated *should* have happened, but *couldn't*). I think this behavior could be preserved with a single bool if we move to your suggested "separate go interface" scheme (i.e. no logging if it doesn't fulfill the interface, logging if it fulfills the interface but returns false)